game/engine.py

```
@@ -219,7 +221,22 @@ def from_fen(cls, fen: str, time_limit=600):
+                col = files.index(parts[3][0])
+                row = 8 - int(parts[3][1])
+                game.en_passant_target = (row, col)
+            except (ValueError, IndexError):
+                pass
+        
         game.halfmove_clock = 0
+        if len(parts) >= 5:
+            try:
+                game.halfmove_clock = int(parts[4])
+            except ValueError:
+                pass
```

Ruff (0.15.14)

[warning] Use `contextlib.suppress(ValueError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(ValueError): ...`

(SIM105)